etcdserver/util.go

```diff
@@ -52,3 +52,34 @@ func numConnectedSince(transport rafthttp.Transporter, since time.Time, self typ
 	}
 	return connectedNum
 }
+
+// longestConnected chooses the member with longest active-since-time.
+// It returns false, if nothing is active.
+func longestConnected(tp rafthttp.Transporter, self types.ID, membs []types.ID) (types.ID, bool) {
```

Comment:
We don't `AddPeer` self, so think it's needed? https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L449-L451